test/functional/tests/fault_injection/test_fault_power_hit.py

```diff
@@ -0,0 +1,86 @@
+import time
+
+from api.cas import casadm, cli, cli_messages
+from api.cas.cache_config import CacheMode
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_utils.size import Size, Unit
+from test_utils.output import CmdException
+
+mount_point = "/mnt/test"
+log_path = "/var/log/opencas.log"
+wait_time_180_s = 180
+wait_time_15_s = 15
+
+
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_fault_power_hit(cache_mode):
+    """
+        title: Test with  power hit.
+        description: |
+          Test if there will be no metadata initialization after wake up
+          - when starting cache without initialization.
+        pass_criteria:
+          - Start cache without re-initialization failed.
+          - Start cache with load works correctly.
+          - Expected message found in log.
+    """
+    with TestRun.step("Prepare CAS device."):
+        cache_disk = TestRun.disks['cache']
+        core_disk = TestRun.disks['core']
+        cache_disk.create_partitions([Size(2, Unit.GibiByte)])
+        core_disk.create_partitions([Size(2, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+        core_dev = core_disk.partitions[0]
+
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+        core = cache.add_core(core_dev)
+        cache_device_link = cache_dev.get_device_link("/dev/disk/by-id")
+
+    with TestRun.step("Hard reset."):
+        power_control = TestRun.plugin_manager.get_plugin('power_control')
+        power_control.power_cycle()
+
+    with TestRun.step("Start cache without re-initialization."):
+        output = TestRun.executor.run_expect_fail(cli.start_cmd(
+            cache_dev=str(cache_dev.system_path),
+            cache_mode=str(cache_mode.name.lower()),
+            force=False, load=False))
+        if cli_messages.check_stderr_msg(output, cli_messages.cache_start_failed):
+            TestRun.LOGGER.info(f"Found expected exception: {cli_messages.cache_start_failed}")
+
+    with TestRun.step("Start cache with load."):
+        try:
+            cache_dev.system_path = cache_device_link.get_target()
+            cache = casadm.load_cache(cache_dev)
+            TestRun.LOGGER.info(f"Cache device loaded correctly (as expected).")
+        except CmdException as e:
+            TestRun.LOGGER.error(f"Failed to load cache device. Exception: {e.output}")
+
+        time.sleep(wait_time_15_s)
+        message_found = check_log(cli_messages.reinitialize_with_force_or_recovery)
+
+        if not message_found:
+            time.sleep(wait_time_180_s)
+            check_log(cli_messages.reinitialize_with_force_or_recovery)
+
+
+def check_log(expected_message):
+    """Read recent lines in log, look for given, expected message."""
+    cmd = f"tail -n 200 {log_path}"
```

Comment:
There may be some entries in the file. In case like that your method will result in a false pass. Try to find a way to read only new entries.

Reply:
good point!